python/fledge/services/core/user_model.py

```diff
@@ -447,13 +451,39 @@ async def delete_token(cls, token):
                 if not ex.error["retryable"]:
                     pass
                 raise ValueError(ERROR_MSG)
-
+            # Remove user session on basis of token
+            await cls.user_session(action="REMOVE", data={"token": token})
             return res
 
         @classmethod
         async def delete_all_user_tokens(cls):
             storage_client = connect.get_storage_async()
             await storage_client.delete_from_tbl("user_logins")
+            await cls.user_session(action="REMOVE")
+
+        @classmethod
+        async def user_session(cls, action, data=None):
```

Comment:
Single def has a reason as we can't declare server import globally, therefore to avoid cyclic import. Otherwise for  every new def we need to import server.

> I see no await statement here.

Not sure where

Reply:
Actually one has return others are not; even if import is duplicate. Splitting will make more clean. Can be a inner class "Sessions" with Sessions.get, save, delete, clear.

Reply:
We have this def as async, but no statement in def is having await?

Reply:
`await cls.user_session(....` used everywhere

Reply:
In definition statements itself;

There may be RuntimeWarning: Coroutine never awaited